SDKAndroid/src/test/java/app/cybrid/sdkandroid/core/BigDecimalTest.kt

```diff
@@ -192,6 +207,144 @@ class BigDecimalTest {
         Assert.assertEquals(value, expected, 0.0)
     }
 
+    @Test(expected = AssertionError::class)
+    fun toDoubleFailureTest() {
+
+        // -- Given
+        val shaNumber = "1157920892373161954235709850086879078532699846656405640394575"
+        val bigDecimalNumber = BigDecimal(shaNumber).setScale(100)
+
+        // -- When
+        val value = bigDecimalNumber.toDouble()
+
+        // -- Then
+        Assert.assertEquals(value, 1.23456789123456794E18, 1.0)
+    }
+
+    @Test
+    fun toDoubleFailureCatchTest() {
+
+        // -- Given
+        val shaNumber = "1157920892373161954235709850086879078532699846656405640394575"
+        val bigDecimalNumber = BigDecimal(shaNumber).setScale(100)
+        var value = 0.0
+        var error: Throwable? = null
+
+        // -- When
+        try {
+            value = bigDecimalNumber.toDouble()
+        } catch (e: Throwable) {
+            error = e
+        }
+
+        // -- Then
+        Assert.assertNull(error?.cause)
+        Assert.assertNull(error?.message)
```

Comment:
Should these be asserting null?

Reply:
yes!! cause the cause and message are null, but the assert it not null

Reply:
`Assert.assertNotNull(error)`